### PR TITLE
Improve GitHub Actions configuration

### PR DESCRIPTION
**Pull request by imbsky**

### CODE REVIEW

**Comment by giltho** [comment]
Oh nice they finally updated checkout v2 to handle submodules !

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        ocaml-version:
+          - 4.09.1
+          - 4.08.1
+          - 4.06.1
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2.1.0
```
Comment (giltho) [inline .github/workflows/build-and-test.yml:27]:
```suggestion
        uses: actions/checkout@v2
```

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        with:
+          submodules: true
+
+      - name: Use latest LTS Node.js release
+        uses: actions/setup-node@v1.4.1
```
Comment (giltho) [inline .github/workflows/build-and-test.yml:32]:
```suggestion
        uses: actions/setup-node@v1
```

File: .github/workflows/build-and-test.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        with:
+          node-version: 12.x
+
+      - name: Use OCaml ${{ matrix.ocaml-version }}
+        uses: avsm/setup-ocaml@v1.0
```
Comment (giltho) [inline .github/workflows/build-and-test.yml:37]:
```suggestion
        uses: avsm/setup-ocaml@v1
```

**Comment by imbsky** [comment]
Yes, that change was a real nuisance to users. We have also been working on improving actions cache recently and I think it will be able to set up OCaml CI environment faster in the near future.

**Comment by imbsky** [comment]
I knew that, but I intentionally choose this versioning method. Well, that sounds reasonable too.

**Comment by giltho** [comment]
> I knew that, but I intentionally choose this versioning method. Well, that sounds reasonable too.

Ah sorry then 😅 I do find the syntax for choosing versions quite unclear though

**Comment by imbsky** [comment]
I agree...

**Comment by giltho** [comment]
argghhhh `@v1` is not tracking anything for `setup-ocaml`  because it's not using semantic versioning (it released `v1.0` instead of `v1.0.0`). So it should indeed be v1.0

**Comment by imbsky** [comment]
It seems that setup-ocaml does not have the v1 tag.

**Comment by imbsky** [comment]
Changed.

**Comment by imbsky** [comment]
Thanks for your review! You and I should have been invited to the setup-ocaml repository as collaborators, so let's make some tweaks to the tag and more. For details, you should have received a message from avsm in Slack.

**Pull request merged**